system/jhtree/ctfile.cpp

```diff
@@ -103,26 +104,27 @@ extern bool isCompressedIndex(const char *filename)
     if (size)
     {
         KeyHdr hdr;
-        if (io->read(0, sizeof(hdr), &hdr) == sizeof(hdr))
+        if (!io->read(0, sizeof(hdr), &hdr) == sizeof(hdr))
```

Comment:
This gives errors on macos builds. I think you need to say 

    if (io->read(0, sizeof(hdr), &hdr) != sizeof(hdr))

Reply:
(see the checks that failed...)

Reply:
oops!

Reply:
Better to regress and not have the unrelated change